transport/tchannel/handler.go

```diff
@@ -192,17 +193,23 @@ type responseWriter struct {
 	buffer             *bufferpool.Buffer
 	response           inboundCallResponse
 	isApplicationError bool
+	originalHeader     bool
 }
 
-func newResponseWriter(response inboundCallResponse, format tchannel.Format) *responseWriter {
+func newResponseWriter(response inboundCallResponse, format tchannel.Format, originalHeader bool) *responseWriter {
 	return &responseWriter{
-		response: response,
-		format:   format,
+		response:       response,
+		format:         format,
+		originalHeader: originalHeader,
 	}
 }
 
 func (rw *responseWriter) AddHeaders(h transport.Headers) {
-	for k, v := range h.Items() {
+	headers := h.Items()
```

Comment:
you should be able to only use `OriginalItems` here, since we're using the `transport.Headers`, so it will canonicalize the headers as required.

you should be able to just do `for k, v := range h.OriginalItems() {`